src/main/java/com/google/api/codegen/transformer/MethodTransformerContext.java

```diff
@@ -79,11 +80,31 @@ public InterfaceConfig getInterfaceConfig() {
   }
 
   public Collection<CollectionConfig> getCollectionConfigs() {
-    return getInterfaceConfig().getCollectionConfigs();
+    return getApiConfig().getCollectionConfigs().values();
   }
 
   public CollectionConfig getCollectionConfig(String entityName) {
-    return getInterfaceConfig().getCollectionConfig(entityName);
+    return getApiConfig().getCollectionConfig(entityName);
+  }
+
+  public boolean entityNameIsCollectionConfig(String entityName) {
+    return getApiConfig().getCollectionConfig(entityName) != null;
+  }
+
+  public boolean entityNameIsCollectionOneofConfig(String entityName) {
+    return getApiConfig().getCollectionOneofConfigs().get(entityName) != null;
+  }
+
+  public CollectionConfig getFirstCollectionConfig(String entityName) {
```

Comment:
This method exists to support method sample generation, which requires any collection config to use as an example. It is only used in one place - maybe it would be clearer to remove the method, and put the logic to get the first collection config in the init transformer.